src/libstd/sys/unix/ext/net.rs

```diff
@@ -90,6 +92,57 @@ unsafe fn sockaddr_un(path: &Path) -> io::Result<(libc::sockaddr_un, libc::sockl
     Ok((addr, len as libc::socklen_t))
 }
 
+unsafe fn recv_vectored_with_ancillary_from(
+    socket: &Socket,
+    bufs: &mut [IoSliceMut<'_>],
+    ancillary: &mut SocketAncillary,
+) -> io::Result<(usize, io::Result<SocketAddr>)> {
+    let mut msg_name: libc::sockaddr_un = mem::zeroed();
+
+    let mut msg = libc::msghdr {
+        msg_name: &mut msg_name as *mut _ as *mut _,
+        msg_namelen: mem::size_of::<libc::sockaddr_un>() as libc::socklen_t,
+        msg_iov: bufs.as_mut_ptr().cast(),
+        msg_iovlen: bufs.len(),
+        msg_control: ancillary.buffer.as_mut_ptr().cast(),
+        msg_controllen: ancillary.buffer.capacity(),
+        msg_flags: 0,
+    };
+
+    let count = socket.recv_msg(&mut msg)?;
+
+    ancillary.buffer.set_len(msg.msg_controllen);
+    ancillary.truncated = msg.msg_flags & libc::MSG_CTRUNC == libc::MSG_CTRUNC;
```

Comment:
Thinking about it, none of the functions currently support detecting whether the main payload was truncated either, right? We could carry that information in `SocketAncillary` too. Or add it to the return tuple.

Reply:
It would be better to add this in the return tuple.

Reply:
Done in cc0b18d3565cb7450716881e7592c008095ab8be